tests/test_core.py

```diff
@@ -633,22 +635,119 @@ def test_chmod(azure):
 def test_chown(azure):
     with azure_teardown(azure):
         azure.touch(a)
-
+        # fake IDs
+        user_id = '470c0ccf-c91a-4597-98cd-48507d2f1486'
+        group_id = '6b190b7a-0acf-43c8-ab14-965f5aea6243'
+        
         # Account doesn't have permission to change owner
         owner = azure.info(a)['owner']
-        azure.chown(a, owner='foo')
+        with pytest.raises(PermissionError):
+            azure.chown(a, owner=user_id)
+        
         assert owner == azure.info(a)['owner']
 
         # Account doesn't have permission to change group
         group = azure.info(a)['group']
-        azure.chown(a, group='bar')
+        with pytest.raises(PermissionError):
+            azure.chown(a, group=group_id)
+        
         assert group == azure.info(a)['group']
 
         # Account doesn't have permission to change owner/group
-        azure.chown(a, owner='foo', group='bar')
+        with pytest.raises(PermissionError):
+            azure.chown(a, owner=owner, group=group)
+        
         assert owner == azure.info(a)['owner']
         assert group == azure.info(a)['group']
 
+@my_vcr.use_cassette
+def test_acl_management(azure):
+    user_id = '470c0ccf-c91a-4597-98cd-48507d2f1486'
+    acl_to_add = 'user:{}:rwx'.format(user_id)
+    acl_to_modify = 'user:{}:-w-'.format(user_id)
+    acl_to_remove = 'user:{}'.format(user_id)
+    with azure_teardown(azure):
```

Comment:
Is `azure` a pytest fixture? If so you should be able to have the fixture yield the value for `azure` and do proper teardown in the fixture itself.